app/src/main/java/org/simple/clinic/sync/indicator/SyncIndicatorViewController.kt

```diff
@@ -148,59 +140,18 @@ class SyncIndicatorViewController @Inject constructor(
   private fun showPendingSyncStatus(events: Observable<UiEvent>): Observable<UiChange> {
     val screenCreates = events.ofType<SyncIndicatorViewCreated>()
 
-    val isPatientSyncPending = screenCreates
+    return screenCreates
```

Comment:
`RxKotlin` has an extension function for `combineLatest` on any `Iterable<Observable<T>>` that combines the output of each source into a `List<T>`. I tried using this to simplify the chain:

```kotlin
return screenCreates
        .flatMap {

          frequentlySyncingRepositories
              .map { it.pendingSyncRecordCount() }
              .combineLatest { counts -> counts.any { it > 0 } }
              .filter { isSyncPending -> isSyncPending }
              .distinctUntilChanged()
              .map { { ui: Ui -> ui.updateState(SyncPending) } }
        }
```

What do you think?

Reply:
This also will let us do `.combineLatest { counts -> counts.sum() > 0 }` which will indicate the total number of pending records.